client/state/jetpack/modules/actions.js

```diff
@@ -149,15 +149,27 @@ export const fetchModuleList = ( siteId ) => {
 					active: module.activated,
 					...omit( module, 'activated' ),
 				} ) );
-
+				const queryClientConnectionData = queryClient.getQueryData( [
+					JETPACK_CONNECTION_HEALTH_QUERY_KEY,
+					siteId,
+				] );
+				if ( queryClientConnectionData?.is_healthy === false ) {
```

Comment:
Could this be a separate reducer? It'll be more straightforward to operate on Redux state within those actions. Passing `queryClient` here seems like mixing two different layers of managing data which we shouldn't. Unless you want to migrate everything to RQ I'd advise to stay in Redux land.

Reply:
Thanks @flootr for your feedback.What `fetchModuleList`  does, is just an api call, and dispatches some actions that are not relevant to our case.
I'm thinking it like that:
![image](https://github.com/Automattic/wp-calypso/assets/497103/a5c2925a-29f2-4cf4-81ae-a840c960d761).
However, if you think it will be better to move away from rq for now, it's totally fine.
I just thought that as a first step to move towards to rq.

Reply:
What @flootr said. And this can get even more complex, and with potential for weird data sync bugs if we continue following the same pattern.